x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -602,12 +608,10 @@ private void computeDecay() {
             long now = threadPool.relativeTimeInMillis();
             for (int i = 0; i < maxFreq; i++) {
                 for (Entry<CacheFileRegion> entry = freqs[i]; entry != null; entry = entry.next) {
-                    if (now - entry.lastAccessed >= 2 * minTimeDelta) {
-                        if (entry.freq > 0) {
-                            unlink(entry);
-                            entry.freq--;
-                            pushEntryToBack(entry);
-                        }
+                    if (entry.freq > 0 && now - entry.lastAccessed >= 2 * minTimeDelta) {
```

Comment:
Bit of vodoo, the `freq` is not volatile, so lets check it first to save some cycles here and there, also the needlessly nested `if` was annoying :)